lib/config/config.js

```diff
@@ -439,10 +454,82 @@ function loadConfigFile (configFile) {
   return null
 }
 
+function printLoggingPreamble (config, sources, configFilePath) {
+  const { logger } = config
+  const nonDefaultKeys = ['environment', 'start', 'file']
+  const nonDefaultOpts = Object.assign.apply(null, nonDefaultKeys.map(k => sources[k]).reverse())
+  const isPreview = AGENT_VERSION.indexOf('-') !== -1
+
+  // Version and pre-release warning
+  logger.info(`Elastic APM Node.js Agent, version: ${AGENT_VERSION}`)
+
+  if (isPreview) {
+    logger.warn('This is a pre-release version and not intended for use in production environments!')
+  }
+
+  // Environment
+  const offsetHours = -((new Date()).getTimezoneOffset() / 60)
+  const offsetSign = offsetHours >= 0 ? '+' : '-'
+  const offsetPad = Math.abs(offsetHours) < 10 ? '0' : ''
+  const timeZone = `UTC${offsetSign}${offsetPad}${Math.abs(offsetHours * 100)}`
+  const frameWorkName = nonDefaultOpts.frameWorkName || 'unknown'
+  const frameWorkVersion = nonDefaultOpts.frameWorkVersion || ''
+  const environmentMessage = [
+    '\nEnvironment:',
+    `- Process ID:  ${process.pid}`,
+    `- Process name: ${process.title}`,
+    // TODO: check sec concerns
+    // ` - command line: ${process.cwd()}`,
+    `- Operating System: ${os.release()}`,
+    `- CPU Architecture: ${os.arch()}`,
+    `- Host: ${os.hostname()}`,
+    `- Time Zone: ${timeZone}`,
+    `- Runtime: Node.js ${process.version}`,
+    // sometimes this info is delayed (maybe not be available)
+    `- Framework: ${frameWorkName} ${frameWorkVersion}`
+  ]
+
+  logger.info(environmentMessage.join('\n'))
+
+  // Config
+  const optionsMessage = ['\nAgent Configuration:']
+
+  if (configFilePath) {
+    optionsMessage.push(
+      '- configuration file used:',
+      `  - ${configFilePath}`
+    )
+  }
+
+  for (const optKey in nonDefaultOpts) {
+    if (EXCLUDE_FIELDS[optKey]) {
+      continue
+    }
+
+    // const origin = nonDefaultKeys.reduce((orig, key) => (optKey in sources[key]) ? key : orig, '')
+    const origin = nonDefaultKeys.find(key => (optKey in sources[key]))
+    let value
+
+    if (REDACT_FIELDS[optKey]) {
+      value = REDACTED
+    } else if (NICE_REGEXPS_FIELDS[optKey] && Array.isArray(nonDefaultOpts[optKey])) {
+      // JSON.stringify() on a RegExp is "{}", which isn't very helpful.
+      value = nonDefaultOpts[optKey].map(r => r instanceof RegExp ? r.toString() : r)
```

Comment:
The regexp fields are all internal processed versions of the config vars, I think, so we shouldn't need to show them in the logging preamble.